File: 이코테/이진탐색/기본코드/binarySearch.py
# 재귀함수 버전
def binary_search(array, target, start, end):
    # 탐색할 곳이 안 남으면 None반환
    if start > end:
        return None
    mid = (start + end) // 2
    # 찾은 경우 중간점 인덱스 반환
    if array[mid] == target:
        return mid
    # 목표값이 중간점의 값보다 작은 경우 왼쪽 확인
    elif array[mid] > target:
        return binary_search(array, target, start, mid - 1)
    # 목표값이 더 크면 오른쪽 확인
    else:
        return binary_search(array, target, mid + 1, end)


n, target = list(map(int, input().split()))

array = list(map(int, input().split()))

result = binary_search(array, target, 0, n - 1)
if result == None:
    print("원소가 없음")
else:
    print(result + 1)

File: 기본 코드/Java/searching/BinarySearch.java
package Java.searching;

import java.util.*;

public class BinarySearch {
	public static int binarySearch(int[] arr, int target, int start, int end) {
		if(start > end) return -1;
		int mid = (start + end) / 2;
		// 재귀를 반복하다 찾은 경우 중간점 인덱스 반환
		if(arr[mid] == target) {
			return mid;
		} else if(arr[mid] > target) {
			return binarySearch(arr, target, start, mid - 1);
		} else {
			return binarySearch(arr, target, mid + 1, end);
		}
	}
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		// 원소의 개수(n)와 찾고자 하는 값(target)을 입력받기 
		int n = sc.nextInt();
		int target = sc.nextInt();
		
		// 전체 원소 입력받기
		int[] arr = new int[n];
		for(int i = 0; i < n; i++) {
			arr[i] = sc.nextInt();
		}
		
		int result = binarySearch(arr, target, 0, n - 1);
		if(result == -1) {
			System.out.println("원소가 존재하지 않습니다.");
		} else {
			System.out.println(result + 1);
		}
		
		sc.close();
	}
}
